diff --git a/python/private/pypi/hub_builder.bzl b/python/private/pypi/hub_builder.bzl
--- a/python/private/pypi/hub_builder.bzl
+++ b/python/private/pypi/hub_builder.bzl
@@ -400,17 +400,7 @@ def _detect_interpreter(self, pip_attr):
             pip_attr.python_version.replace(".", "_"),
         )
         if python_name not in self._available_interpreters:
-            fail((
-                "Unable to find interpreter for pip hub '{hub_name}' for " +
-                "python_version={version}: Make sure a corresponding " +
-                '`python.toolchain(python_version="{version}")` call exists.' +
-                "Expected to find {python_name} among registered versions:\n  {labels}"
-            ).format(
-                hub_name = self.name,
-                version = pip_attr.python_version,
-                python_name = python_name,
-                labels = "  \n".join(self._available_interpreters),
-            ))
+            return None
         python_interpreter_target = self._available_interpreters[python_name]
 
     return struct(
