--- BUILD.bazel
+++ BUILD.bazel
@@ -403,18 +403,18 @@
 cc_shared_library(
     name = "curl_shared",
-    additional_linker_inputs = [
-        ":symbol_versioning_map",
-    ],
+    # additional_linker_inputs = [
+    #     ":symbol_versioning_map",
+    # ],
     shared_lib_name = select({
         "@platforms//os:macos": "libcurl.dylib",
         "@platforms//os:windows": "libcurl.dll",
         "//conditions:default": "libcurl.so",
     }),
     user_link_flags = select({
         "@platforms//os:windows": [],
         "@platforms//os:macos": [],
         "//conditions:default": [
-            "-Wl,--version-script=$(location :symbol_versioning_map)",
+            # "-Wl,--version-script=$(location :symbol_versioning_map)",
         ],
     }),
     visibility = ["//visibility:public"],
