diff --git a/xla/tsl/util/filewrapper.cc b/xla/tsl/util/filewrapper.cc
--- a/xla/tsl/util/filewrapper.cc
+++ b/xla/tsl/util/filewrapper.cc
@@ -354,9 +354,11 @@
   initializers.reserve(infiles.size());
 
   // Copied from base/port.h and renamed.
-  f_cc << "#if defined(COMPILER_MSVC)\n";
+  f_cc << "#if defined(COMPILER_MSVC) || defined(_MSC_VER)\n";
   f_cc << "#define ALIGN_ATTRIBUTE(X) __declspec(align(X))\n";
   f_cc << "#elif defined(__GNUC__) || defined(COMPILER_ICC)\n";
   f_cc << "#define ALIGN_ATTRIBUTE(X) __attribute__((aligned(X)))\n";
+  f_cc << "#else\n";
+  f_cc << "#define ALIGN_ATTRIBUTE(X)\n";
   f_cc << "#endif\n";
 
   f_cc << "namespace {\n";
