diff --git a/third_party/mkl_dnn/mkldnn_v1_async.BUILD b/third_party/mkl_dnn/mkldnn_v1_async.BUILD
--- a/third_party/mkl_dnn/mkldnn_v1_async.BUILD
+++ b/third_party/mkl_dnn/mkldnn_v1_async.BUILD
@@ -167,7 +167,10 @@
     "src/cpu/gemm",
     "src/graph",
     "third_party",
-] + if_sycl_build_is_configured(
+] + select({
+    "@xla//xla/tsl:aarch64_or_cross": ["third_party/xbyak_aarch64/xbyak_aarch64"],
+    "//conditions:default": [],
+}) + if_sycl_build_is_configured(
     [
         "include/oneapi",
         "include/oneapi/dnnl",
@@ -242,7 +245,10 @@
 # compiler doesn't clean up stack from temporary objects.)
 cc_library(
     name = "onednn_autogen",
-    srcs = glob(["src/cpu/x64/gemm/**/*_kern_autogen*.cpp"]),
+    srcs = select({
+        "@xla//xla/tsl:aarch64_or_cross": [],
+        "//conditions:default": glob(["src/cpu/x64/gemm/**/*_kern_autogen*.cpp"]),
+    }),
     copts = [
         "-O1",
         "-U_FORTIFY_SOURCE",
@@ -268,7 +274,6 @@
             "src/cpu/*.cpp",
             "src/cpu/**/*.cpp",
             "src/cpu/jit_utils/**/*.cpp",
-            "src/cpu/x64/**/*.cpp",
             "src/graph/interface/*.cpp",
             "src/graph/backend/*.cpp",
             "src/graph/backend/dnnl/*.cpp",
@@ -285,10 +290,25 @@
             "src/cpu/aarch64/**",
             "src/cpu/ppc64/**",
             "src/cpu/rv64/**",
-            "src/cpu/x64/gemm/**/*_kern_autogen.cpp",
+            "src/cpu/x64/**",
             "src/cpu/sycl/**",
         ],
-    ),
+    ) + select({
+        "@xla//xla/tsl:aarch64_or_cross": glob(
+            ["src/cpu/aarch64/**/*.cpp"],
+            exclude = [
+                "src/cpu/aarch64/acl_*.cpp",
+                "src/cpu/aarch64/**/acl_*.cpp",
+            ],
+        ) + [
+            "third_party/xbyak_aarch64/src/xbyak_aarch64_impl.cpp",
+            "third_party/xbyak_aarch64/src/util_impl.cpp",
+        ],
+        "//conditions:default": glob(
+            ["src/cpu/x64/**/*.cpp"],
+            exclude = ["src/cpu/x64/gemm/**/*_kern_autogen.cpp"],
+        ),
+    }),
     copts = _COPTS_LIST + [
         "-DDNNL_ENABLE_ITT_TASKS",  # Enable ITT for CPU
     ],
@@ -319,7 +339,6 @@
             "src/cpu/**/*.cpp",
             "src/common/ittnotify/*.c",
             "src/cpu/jit_utils/**/*.cpp",
-            "src/cpu/x64/**/*.cpp",
             "src/graph/interface/*.cpp",
             "src/graph/backend/*.cpp",
             "src/graph/backend/dnnl/*.cpp",
@@ -345,14 +364,29 @@
             "src/cpu/aarch64/**",
             "src/cpu/ppc64/**",
             "src/cpu/rv64/**",
-            "src/cpu/x64/gemm/**/*_kern_autogen.cpp",
+            "src/cpu/x64/**",
             "src/cpu/sycl/**",
             "src/common/ittnotify/**",
             "src/gpu/amd/**",
             "src/gpu/intel/conv/jit/v2/planner/planner_main.cpp",
             "src/gpu/nvidia/**",
         ],
-    ) + [
+    ) + select({
+        "@xla//xla/tsl:aarch64_or_cross": glob(
+            ["src/cpu/aarch64/**/*.cpp"],
+            exclude = [
+                "src/cpu/aarch64/acl_*.cpp",
+                "src/cpu/aarch64/**/acl_*.cpp",
+            ],
+        ) + [
+            "third_party/xbyak_aarch64/src/xbyak_aarch64_impl.cpp",
+            "third_party/xbyak_aarch64/src/util_impl.cpp",
+        ],
+        "//conditions:default": glob(
+            ["src/cpu/x64/**/*.cpp"],
+            exclude = ["src/cpu/x64/gemm/**/*_kern_autogen.cpp"],
+        ),
+    }) + [
         ":header_generator",
         ":kernel_list_generator",
     ],
