diff --git a/MODULE.bazel b/MODULE.bazel
index c8ebbce9d2..82178022ae 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -26,7 +26,7 @@ bazel_dep(name = "re2", version = "2025-11-05.bcr.1", repo_name = "com_googlesou
 bazel_dep(name = "rules_cc", version = "0.2.17")
 bazel_dep(name = "rules_java", version = "8.16.1")
 bazel_dep(name = "rules_license", version = "1.0.0")
-bazel_dep(name = "rules_python", version = "1.8.5")
+bazel_dep(name = "rules_python", version = "2.2.0")
 bazel_dep(name = "rules_shell", version = "0.6.1")
 bazel_dep(name = "snappy", version = "1.2.1")
 bazel_dep(name = "zlib", version = "1.3.2")
@@ -153,12 +153,10 @@ single_version_override(
     module_name = "rules_python",
     patch_strip = 1,
     patches = [
-        "//third_party/py:rules_python_pip_version.patch",
         "//third_party/py:rules_python_freethreaded.patch",
-        "//third_party/py:rules_python_versions.patch",
         "//third_party/py:rules_python_scope.patch",
     ],
-    version = "1.8.5",
+    version = "2.2.0",
 )
 
 python = use_extension("@rules_python//python/extensions:python.bzl", "python")
diff --git a/third_party/py/python_init_rules.bzl b/third_party/py/python_init_rules.bzl
index 7db7e94c0d..751194354a 100644
--- a/third_party/py/python_init_rules.bzl
+++ b/third_party/py/python_init_rules.bzl
@@ -27,13 +27,11 @@ def python_init_rules(extra_patches = []):
 
     tf_http_archive(
         name = "rules_python",
-        sha256 = "8964aa1e7525fea5244ba737458694a057ada1be96a92998a41caa1983562d00",
-        strip_prefix = "rules_python-1.8.5",
-        urls = tf_mirror_urls("https://github.com/bazelbuild/rules_python/releases/download/1.8.5/rules_python-1.8.5.tar.gz"),
+        sha256 = "e11d2e1efce1589e5bdfa93986712c74fc7467a0f093143d489d2ef5ebb1ed2a",
+        strip_prefix = "rules_python-2.2.0",
+        urls = tf_mirror_urls("https://github.com/bazelbuild/rules_python/releases/download/2.2.0/rules_python-2.2.0.tar.gz"),
         patch_file = [
-            "@xla//third_party/py:rules_python_pip_version.patch",
             "@xla//third_party/py:rules_python_scope.patch",
             "@xla//third_party/py:rules_python_freethreaded.patch",
-            "@xla//third_party/py:rules_python_versions.patch",
         ] + extra_patches,
     )
diff --git a/third_party/py/rules_python_freethreaded.patch b/third_party/py/rules_python_freethreaded.patch
index d564e9a60a..715c72e026 100644
--- a/third_party/py/rules_python_freethreaded.patch
+++ b/third_party/py/rules_python_freethreaded.patch
@@ -1,3 +1,28 @@
+diff --git a/python/private/pypi/platform.bzl b/python/private/pypi/platform.bzl
+index 00000000..00000000 100644
+--- a/python/private/pypi/platform.bzl
++++ b/python/private/pypi/platform.bzl
+@@ -25,7 +25,7 @@ def platform(*, name, arch_name, os_name, config_settings = [], env = {}, marker
+         # the lowest priority one needs to be the first one
+         whl_platform_tags = ["any"] + whl_platform_tags
+
+-    whl_abi_tags = whl_abi_tags or ["abi3", "cp{major}{minor}"]
++    whl_abi_tags = whl_abi_tags or ["abi3", "cp{major}{minor}", "cp{major}{minor}t"]
+     if "none" not in whl_abi_tags:
+         # the lowest priority one needs to be the first one
+         whl_abi_tags = ["none"] + whl_abi_tags
+diff --git a/python/private/pypi/pep508_env.bzl b/python/private/pypi/pep508_env.bzl
+index 00000000..00000000 100644
+--- a/python/private/pypi/pep508_env.bzl
++++ b/python/private/pypi/pep508_env.bzl
+@@ -23,6 +23,7 @@ _DEFAULT = "//conditions:default"
+ # Here we store the aliases in the platform so that the users can specify any valid target in
+ # there.
+ _cpu_aliases = {
++    "amd64": "x86_64",
+     "arm": "aarch32",
+     "arm64": "aarch64",
+ }
 diff --git a/python/private/python_register_toolchains.bzl b/python/private/python_register_toolchains.bzl
 index 9e75c419..ba6d8f7a 100644
 --- a/python/private/python_register_toolchains.bzl
