--- third_party/extensions/python_version.bzl
+++ third_party/extensions/python_version.bzl
@@ -8,6 +8,8 @@
 HERMETIC_PYTHON_VERSION = "{version}"
 HERMETIC_PYTHON_VERSION_KIND = "{py_kind}"
 USE_PYWRAP_RULES = {use_pywrap_rules}
+# TODO(pcloudy): Figure out how to support requirements_lock in Bzlmod.
+REQUIREMENTS = "//build:requirements_lock_3_11.txt"
 """

 def _python_version_repo_impl(repository_ctx):
